packages/esm-commons-lib/src/components/encounter-list/encounter-list.component.tsx

```diff
@@ -198,6 +200,7 @@ export const EncounterList: React.FC<EncounterListProps> = ({
                     workspaceWindowSize,
                   );
                 }}
+                disabled={actionItem.label == 'Edit form' && disableEdit == true && findEncounterLatestDateIndex(encounters) != encounterIndex}
```

Comment:
This logic looks very specific and doesn't look like something that can be reused because

1. The condition to disable the action seems to be based off the words `Edit form` which would break the moment the text changes
2. This doesn't play well with scenarios where actions are from multiple forms

My suggestion would be moving this logic away from here and adding it to the actions with the individual programs.